src/palace_tools/cli/import_libraries_from_csv.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import base64
+import csv
+import traceback
+from pathlib import Path
+from typing import Any
+
+import httpx
+import typer
+
+from palace.manager.api.admin.controller.library_settings import LibraryImportInfo
+from palace.manager.util.log import pluralize
+
+from palace_tools.utils.http.async_client import HTTPXAsyncClient
+from palace_tools.utils.typer import run_typer_app_as_main
+
+app = typer.Typer()
+
+
+async def upload_csv_rows_to_palace(
+    base_url: str,
+    access_token: str | None,
+    libraries: list[LibraryImportInfo],
+    verbose: bool = False,
+) -> httpx.Response:
+    """Upload CSV rows as JSON to Palace import-libraries-from-csv endpoint."""
+    import_url = f"{base_url.rstrip('/')}/admin/libraries/import"
+
+    headers = {
+        "Authorization": f"Basic {access_token}",
+        "Content-Type": "application/json",
+    }
+
+    if verbose:
+        library_count = len(libraries)
+        typer.echo(
+            f"Importing {len(libraries)} {pluralize(library_count,'library')} to "
+            f"{import_url}...This operation may take any where for a couple of seconds to several minutes "
+            f"depending on how many libraries you are importing."
+        )
+
+    async with HTTPXAsyncClient() as client:
+        # TODO:  The large timeout is necessary because it appears that it takes about 6 seconds per library.
+        #  The main time sink appears to be the creation of the default lanes.
+        #  routine.  If a hundred libraries are imported at once, it could take a
+        #  few minutes, especially if running against a live instance with collections already configured.
+        #  If this time lag becomes a problem, we will likely need to 1) optimize the default lane setup and/or 2) post
+        #  one library at a time rather than trying to import all in one transaction.
+        response = await client.post(
+            import_url,
+            json={"libraries": [library.__dict__ for library in libraries]},
+            headers=headers,
```

Comment:
You shouldn't manually base64 encode the auth parameters, you need to check things like encoding of :, and there is no reason to reinvent the wheel, just use httpx built in auth handling: https://www.python-httpx.org/quickstart/#authentication

```
auth=("my_user", "password123")
```